ELVM.md

```diff
@@ -0,0 +1,68 @@
+# Esoteric Language Virtual Machine
+
+## Registers and memory
+
+* Words are unsigned integers. The word-size is backend dependent, but
+  most backends use 24-bit words.
+* 6 registers: A, B, C, D, SP, and BP. They are one word wide and
+  initialized to zero.
+* Memory addresses are one word wide. The beginning of memory is
+  initialized from the .data segments; the rest is initialized to
+  zero.
+* Instructions are not stored in memory. Every instruction in a basic
+  block has the same pc (program counter) value, and a branch to a pc
+  goes to the first instruction with that pc.
+
+## Ops
+
+MOV dst, src
+- copy src to dst
+- src: immediate or register
+- dst: register
+
+ADD dst, src
+- add src to dst and places result in dst
+- src: immediate or register
+- dst: register
+- no carry flag
+
+SUB dst, src
+- subtract src from dst and places result in dst
+- src: immediate or register
+- dst: register
+- no borrow flag
+
+LOAD dst, src
+- copy address src to dst
+- src: immediate or register
+- dst: register
+
+STORE dst, src
```

Comment:
Very true. Thanks for pointing this out. I think ideally it'd be nicer to correct the internal implementation, but it's somewhat involved as there're already a bunch of backends... For the meantime, I've added a note on the discrepancy: https://github.com/shinh/elvm/commit/34e1ebb8b21e509cf1cfd71d05592516b46ce360